__tests__/serverjs/cubefn.test.js

```diff
@@ -204,3 +210,107 @@ test('addAutocard correctly replaces autocard format strings', () => {
     expect(result).toBe(expected);
   });
 });
+
+describe('CSVtoCards', () => {
+  it('can find a card', async () => {
+    const expectedId = 'aaae15dd-11b6-4421-99e9-365c7fe4a5d6';
+    const expectedCard = {
+      name: 'Embercleave',
+      cmc: '3',
+      type_line: 'Creature - Test',
+      colors: ['U'],
+      set: 'ELD',
+      collector_number: '359',
+      status: 'Owned',
+      finish: 'Is Foil',
+      imgUrl: 'http://example.com/',
+      tags: ['tag1', 'tag2'],
+    };
+    const expectedMaybe = {
+      name: 'Embercleave',
+      cmc: '2',
+      type_line: 'Creature - Type',
+      colors: ['R', 'W'],
+      set: 'ELD',
+      collector_number: '120',
+      status: 'Not Owned',
+      finish: 'Is Not Foil',
+      imgUrl: null,
+      tags: ['tag3', 'tag4'],
+    };
+    const cards = [
+      `"${expectedCard.name}",${expectedCard.cmc},${expectedCard.type_line.replace(
+        '—',
+        '-',
+      )},${expectedCard.colors.join('')},${expectedCard.set},${expectedCard.collector_number},${expectedCard.status},${
+        expectedCard.finish
+      },false,${expectedCard.imgUrl},"${expectedCard.tags.join(',')}"`,
+      `"${expectedMaybe.name}",${expectedMaybe.cmc},${expectedMaybe.type_line.replace(
+        '—',
+        '-',
+      )},${expectedMaybe.colors.join('')},${expectedMaybe.set},${expectedMaybe.collector_number},${
+        expectedMaybe.status
+      },${expectedMaybe.finish},true,undefined,"${expectedMaybe.tags.join(',')}"`,
+    ];
+    await carddb.initializeCardDb(fixturesPath, true);
+    const { newCards, newMaybe, missing } = cubefn.CSVtoCards(cards, carddb);
+    expect.extend({
+      equalsArray(received, expected) {
+        return {
+          message: () => `expected ${received} to equal array ${expected}`,
+          pass: arraysEqual(received, expected),
+        };
+      },
+    });
+    const expectSame = (card, expected) => {
+      expect(card.cardID).toBe(expectedId);
+      expect(card.name).toBe(expected.name);
+      expect(card.cmc).toBe(expected.cmc);
+      expect(card.colors).equalsArray(expected.colors);
+      expect(card.set).toBe(expected.set);
+      expect(card.collector_number).toBe(expected.collector_number);
+      expect(card.status).toBe(expected.status);
+      expect(card.finish).toBe(expected.finish);
+      expect(card.imgUrl).toBe(expected.imgUrl);
+      expect(card.tags).equalsArray(expected.tags);
+    };
+    expect(newCards.length).toBe(1);
+    expectSame(newCards[0], expectedCard);
+    expect(newMaybe.length).toBe(1);
+    expectSame(newMaybe[0], expectedMaybe);
+    expect(missing).toBe('');
+  });
+});
+
+describe('compareCubes', () => {
+  it('can calculate the diff between two cubes', async () => {
+    await carddb.initializeCardDb(fixturesPath, true);
```

Comment:
Might be better in a beforeAll block?

Reply:
I was trying to follow the existing pattern, and I think there are some tests that don't use it currently, like `test('getBasics returns the expected set of basic lands', () => {` from line 117.